sdk/core/azure-core-management/src/main/java/com/azure/core/management/exception/ManagementError.java

```diff
@@ -105,8 +126,82 @@ public List<AdditionalInfo> getAdditionalInfo() {
         return additionalInfo;
     }
 
+    void setAdditionalInfo(List<AdditionalInfo> additionalInfo) {
+        this.additionalInfo = additionalInfo;
+    }
+
     @Override
     public String toString() {
         return message == null ? super.toString() : message;
     }
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        return jsonWriter.writeStartObject()
+            .writeStringField("code", code)
+            .writeStringField("message", message)
+            .writeStringField("target", target)
+            .writeArrayField("details", details, JsonWriter::writeJson)
+            .writeArrayField("additionalInfo", additionalInfo, JsonWriter::writeJson)
+            .writeEndObject();
+    }
+
+    /**
+     * Reads a JSON stream into a {@link ManagementError}.
+     *
+     * @param jsonReader The {@link JsonReader} being read.
+     * @return The {@link ManagementError} that the JSON stream represented, or null if it pointed to JSON null.
+     * @throws IOException If a {@link ManagementError} fails to be read from the {@code jsonReader}.
+     */
+    public static ManagementError fromJson(JsonReader jsonReader) throws IOException {
+        return jsonReader.readObject(reader -> {
+            // Buffer the next JSON object as ResponseError can take two forms:
+            //
+            // - A ResponseError object
+            // - A ResponseError object wrapped in an "error" node.
+            JsonReader bufferedReader = reader.bufferObject();
+            bufferedReader.nextToken(); // Get to the START_OBJECT token.
+            while (bufferedReader.nextToken() != JsonToken.END_OBJECT) {
+                String fieldName = bufferedReader.getFieldName();
+                bufferedReader.nextToken();
+
+                if ("error".equals(fieldName)) {
```

Comment:
Just curious. In `PollOperationDetails`, we already read the value object of the field `error`.
https://github.com/Azure/azure-sdk-for-java/blob/6c4af651bda1ce4b3b81ebce67c2466fec9a0650/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollOperationDetails.java#L83-L85
Here, we read the value object of field `error` again.
Does this mean there will be two nested "error" fields in an Object?
```json
{
    "error": {
        "error": {
        }
    }
}
```

Reply:
It's possible we could see that form of JSON but this was more to handle cases when the error is the outer type where some services return:

```json
{
  "error": {
    "code": "123",
    "message": "Problem",
    ...
  }
}
```

And others return:

```json
{
  "code": "123",
  "message": "Problem",
  ...
}
```

The deserialization logic will correctly handle both. `PollOperationDetails` having an `error` property will also be able to handle both cases as well without issue.

Reply:
So far, `PollOperationDetails` should only used by data-plane. AFAIK, none of the mgmt-plane poll response would take this shape, and we would not generate `Poller<PollOperationDetails, T>` for mgmt-plane.